jmix-core/core/src/main/java/io/jmix/core/security/UserManager.java

```diff
@@ -29,25 +29,54 @@
 public interface UserManager {
 
     /**
-     * Changes the password for the specific user.
+     * Changes the password for the specific user. The changes will be saved to the database.
      *
-     * @param userName    - users login
-     * @param oldPassword - non encoded old users password
-     * @param newPassword - non encoded new users password
-     * @throws PasswordNotMatchException if the oldPassword is not null and the oldPassword does not equal to the current password or
-     *                                   if the oldPassword is null and the newPassword does equal the current password
+     * @param userName    user login
+     * @param oldPassword non-encoded old user password
+     * @param newPassword non-encoded new user password
+     * @throws PasswordNotMatchException if the oldPassword is not {@code null} and
+     *                                   the oldPassword does not equal to the current password or
+     *                                   if the oldPassword is {@code null}
+     *                                   and the newPassword does equal the current password
      */
-    void changePassword(String userName, @Nullable String oldPassword, @Nullable String newPassword) throws PasswordNotMatchException;
-
+    default void changePassword(String userName, @Nullable String oldPassword, @Nullable String newPassword)
+            throws PasswordNotMatchException {
+        changePassword(userName, oldPassword, newPassword, true);
+    }
 
     /**
      * Changes the password for the specific user.
      *
-     * @param users - users which need reset passwords
-     * @return map which contains new password for specific user
+     * @param userName    users login
+     * @param oldPassword non-encoded old user password
+     * @param newPassword non-encoded new user password
+     * @param saveChanges whether to save changes to the database
+     * @throws PasswordNotMatchException if the oldPassword is not {@code null}
+     *                                   and the oldPassword does not equal to the current password or
+     *                                   if the oldPassword is {@code null}
+     *                                   and the newPassword does equal the current password
      */
-    Map<UserDetails, String> resetPasswords(Set<UserDetails> users);
+    UserDetails changePassword(String userName, @Nullable String oldPassword, @Nullable String newPassword,
+                               boolean saveChanges) throws PasswordNotMatchException;
 
+    /**
+     * Changes the password for the passed users. The changes will be saved to the database.
```

Comment:
Generates new passwords for passed users and saves changes to the database immediately